### DISCUSSION TITLE
Question About the Calculation of `code_lens` in MeshTransformer

### DISCUSSION THREAD

**Discussion by qixuema** [lucidrains/meshgpt-pytorch#15] [Q&A]
Hello Phil,

I have been exploring the code in the repository and came across a line that piqued my curiosity regarding the calculation of `code_lens`. Specifically, the [line](https://github.com/lucidrains/meshgpt-pytorch/blob/83e2c0089fec4a41ba8eb47a7d6a9cce9e06b7eb/meshgpt_pytorch/meshgpt_pytorch.py#L1195) is:

```python
code_lens = ((codes != self.pad_id).cumsum(dim = -1) == 0).sum(dim = -1)
```

As I understand it, this line appears to calculate the number of non-padding elements in `codes`. However, I noticed that right after this, `code_lens` is used to set the `eos_token_id`. 
```python
codes[batch_arange, code_lens] = self.eos_token_id
```
Based on the above calculation method, it seems to add the `eos_token_id` at the very beginning of the `codes` sequence, rather than at the end. This seems a bit counterintuitive to me, as typically, `eos_token_id` is added at the end of a sequence.

Since I'm not very familiar with this part of the functionality, I'm curious about the rationale and logic behind this particular calculation method. Is there a specific reason or scenario for this approach, or am I misunderstanding the code?

Looking forward to your clarification, and thank you for your contributions to this project!

Best regards,
Xueqi

**Comment by lucidrains** [comment]
@qixuema hi Xueqi

yes, i do believe that is a bug 🐛 , thank you for catching it! the latest commit should be more sound
